example-client/README.md

```diff
@@ -3,11 +3,219 @@ Example CoAP client
+```
+
+## Java Keystore
+
+#### Create a PKCS #12 archive to store cryptography objects so we can create a Java keystore from it
+
+```
+openssl pkcs12 -export -in lwm2m_device_cert.pem -inkey lwm2m_device_private_key.pem -out my_lwm2m_device_keystore_certs.p12 -name mbed-cloud-${ENDPOINT_NAME} -password pass:secret
+```
+
+#### Create Java keystore importing the PKCS #12 archive
+
+```
+keytool -keystore example-client-device-keystore.jks -storepass secret -importkeystore -srcstorepass secret -srckeystore my_lwm2m_device_keystore_certs.p12 -srcstoretype PKCS12 -noprompt
+```
+
+#### Import our CA cert we created above to the Java keystore
+
+```
+keytool -keystore example-client-device-keystore.jks -storepass secret -import -alias lwm2m_server_ca_certificate -file lwm2m_server_ca_certificate.pem -noprompt
+```
+
+## Convience Commands Viewing/Verifying Keys, Certs, Keystore
+
+### Dump the contents of the certs we created for convience
+
+```
+openssl pkcs12 -in my_lwm2m_device_keystore_certs.p12 -nodes -passin pass:secret | openssl x509 -noout -subject
+openssl x509 -in my_ca_public_cert.pem -noout -text | grep Subject:
+openssl x509 -in lwm2m_server_ca_certificate.pem -noout -text | grep Subject:
+openssl x509 -in lwm2m_device_cert.der -inform der -text -noout
+```
+
+### Examine the keystore
+
+```
+keytool -list -v -storepass secret -keystore example-client-device-keystore.jks
+```
+
+## Mbed Cloud Certificate Uploading
+
+### Output a hash of our private key plus account nexus in base64 for mbedcloud verification
+
+Note: You only need this if you are uploading the cert with using the Portal UI
+
+```
+printf "${ACCOUNT_ID}" > account_id && openssl dgst -sha256 -sign my_ca_private_key.pem -out sign.sha256 account_id && base64 sign.sha256
+```
+
+### Get a json payload file to upload the results with cURL if you didn't do this via Portal
+
+```
+CERT=$(cat my_ca_public_cert.pem | sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/\\n/g')
+
+echo -n "{\"name\": \"Dexter Fryar my_ca_public_cert.pem\", \"description\": \"LWM2M self signed certificate for testing https://github.com/ARMmbed/java-coap/example-client\", \"certificate\": \"${CERT}\", \"service\": \"lwm2m\"}" > payload.json
+```
+
+### Upload the certificate to Mbed Cloud if you didn't use the portal
+
+```
+curl -X POST \
+  -d @payload.json \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer '"${API_KEY}"'' \
+  -H 'Cache-Control: no-cache' \
+  https://api.us-east-1.mbedcloud.com/v3/trusted-certificates
+```
+
+note: Using the [Portal UI](https://portal.us-east-1.mbedcloud.com/identity/certificates) you can verify the certificate is now in place
+
+<img src="images/certificates.png" width="600">
+
+# Troubleshooting
+
+### Known Issues or Workarounds
+
+[Timeout trying to connect to arm mbed remote server](https://github.com/ARMmbed/java-coap/issues/3)
```

Comment:
We should not be using the title case (capitalizing the words in heading) -> Known issues and workarounds